### DISCUSSION TITLE
leetcode 의 analyze complexity 를 믿어도 될까요?

### DISCUSSION THREAD

**Discussion by wogha95** [DaleStudy/leetcode-study#365] [도움요청] [answered]
[Construct Binary Tree From Preorder And Inorder Traversal](https://github.com/DaleStudy/leetcode-study/issues/253) 문제를 풀고 complexity 계산 도중에 의문점이 생겼습니다.

[제 풀이](https://github.com/DaleStudy/leetcode-study/pull/339/files#diff-03d92d3baacd17bb0c5fb2f2e947f8fe27fc7efd9176b989f72e12811b86199a)의 시간복잡도는 preorder 순회 `O(N)` 에 findIndex 순회 `O(N)` 이 중첩되어 있으니 `O(N^2)`로 생각했습니다. 그러나 leetcode analyze 에서는 시간복잡도를 `O(N)`으로 알려줍니다. 혹시 제가 시간복잡도 계산을 잘못한 것일까요?

**Comment by SamTheKorean** [comment]
해당 문제를 아직 보진 않았지만, 일반적으로 분석이 맞지만 틀린 경우나 매우 창의적으로 분석하는 경우도 종종 봐서 개인적으로 신뢰하지는 않고 있습니다!

**Comment by DaleSeo** [comment, marked as answer]
저도 LeetCode의 복잡도 분석 기능을 써봤는데 틀린 적이 너무 많아서 신뢰하고 있지 않습니다. 직접 분석하신 결과 `O(n^2)`이 맞다고 생각합니다 👍

**Comment by TonyKim9401** [comment]
전 TC, SC 모두 제가 계산 후에 검토용으로 코드 그대로 입력해서 GPT로 계산 해달라고 하고 비교해봐요 :)
